spec/takeaway_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'takeaway'
+require 'menu'
+require 'basket'
+describe TakeAway do
+  subject(:takeaway) { described_class.new(menu, basket, sms, config: {}) }
+
+  let(:menu) { instance_double "Menu", generator: user_list, list: list }
+  let(:user_list) { double :user_list }
+  let(:list) { double :list }
+  let(:list_number) { double :list_number }
+  let(:value) { double :value }
+  let(:dish) { double :dish }
+  let(:basket) { instance_double("Basket") }
+  let(:sms) { double :sms, deliver: nil }
```

Comment:
You have a lot of classes, so you have a lot of lets in your rspec. Maybe slightly over engineered, but very good code overall.